### DISCUSSION TITLE
How can I make curves/vertices push them away from each other?

### DISCUSSION THREAD

**Discussion by gerroon** [nortikin/sverchok#4678] [Q&A]
Hi

Is there a way to push curves or vertices of objects push them each other away so that they maintain a minimal space between them? I would like my curves to not touch eachother.

thanks

**Comment by portnov** [comment]
Can you draw an example of what do you want to get?

**Reply by gerroon** [reply]
> Can you draw an example of what do you want to get?

Sure. Please see the image. As you can see the curve tubes are overlapping. I want these curves to avoid or push each other so that they do not overlap. I do not necessarily care about how they clear up the space between them. I tried  the Pulga stuff but I was not able to do it properly, I am guessing that I do not understand it properly maybe there is a better way in Svberchok/

This is about modifying existing geometry rather than creating new curves from scratch.

![blender_twRYbbxeJE](https://user-images.githubusercontent.com/8519469/191661001-00c40ac6-6de8-4d28-821b-37394300f5ea.jpg)

**Comment by portnov** [comment]
From the top of my head, I don't have an idea of how to formulate exactly the idea of "pushing curves from one another". But I have an idea about how to push curves from some point.
![Screenshot_20220922_214807](https://user-images.githubusercontent.com/284644/191806031-e8e478b0-c548-463d-88b3-8ed3d667021d.png)
[push_curves_t1_2022_09_22_16_48.zip](https://github.com/nortikin/sverchok/files/9627837/push_curves_t1_2022_09_22_16_48.zip)

**Comment by gerroon** [comment]
@portnov  Thanks for idea. I think this would work if all I have the deal with is a  single point of tangling. I have curves that overlap in many places, so I am trying to figure out an automated way of minimizing overlaps.

**Comment by gerroon** [comment]
Houdini has this node called detangle, so I am looking something along those lines at least.
